File: app/src/main/java/com/univreview/adapter/contract/ReviewListAdapterContract.java
package com.univreview.adapter.contract;

import com.univreview.listener.OnItemClickListener;
import com.univreview.model.model_kotlin.AbstractDataProvider;

/**
 * Created by DavidHa on 2017. 7. 14..
 */

public interface ReviewListAdapterContract {
    interface View{
        void setOnItemClickListener(OnItemClickListener itemClickListener);
    }

    interface Model{
        void addItem(AbstractDataProvider item);
        void clearItem();
        AbstractDataProvider getItem(int position);
    }
}

File: app/src/main/java/com/univreview/adapter/contract/ReviewListAdapterContract.kt
package com.univreview.adapter.contract

import com.dhha22.bindadapter.BindAdapterContract
import com.dhha22.bindadapter.listener.OnItemClickListener

/**
 * Created by DavidHa on 2017. 9. 1..
 */
interface ReviewListAdapterContract {
    interface View {
        fun setOnItemClickListener(itemClickListener: OnItemClickListener)
        fun setMoreItemClickListener(itemClickListener: OnItemClickListener)
    }

    interface Model : BindAdapterContract.Model
}